### PR TITLE
add setting options

### PR DESCRIPTION
**Pull request by ZmagoD**
Screenshot 2020-08-25 at 23 11 56

@reichert621 I added the settings in the general panel. Now we have a working version of the plugin

### CODE REVIEW

File: admin/class-papercups-admin.php [hunk 1]
```diff
@@ -83,51 +70,175 @@ public function enqueue_styles()
...
+    register_setting('general', 'papercups_greeting');
+    register_setting('general', 'papercups_primary_color');
+    register_setting('general', 'papercups_require_email_upfront');
+    register_setting('general', 'papercups_base_url', array('default' => 'https://app.papercups.io'));
+  }
+
+  private function add_settings_fields() {
+    add_settings_field(
+      'papercups_account_id',
+      __('Papercups accound ID', $this->plugin_name),
```
Comment (reichert621) [inline admin/class-papercups-admin.php:106]: small typo: `accound` -> `account`

File: admin/class-papercups-admin.php [hunk 2]
```diff
@@ -83,51 +70,175 @@ public function enqueue_styles()
...
+      array($this, 'text_input_callback'),
+      'general',
+      'papercups_settings_section',
+      array(
+        'type'      => 'text',
+        'value'     => get_option('papercups_account_id'),
+        'class'     => 'regular-text ltr',
+        'label_for' => 'papercups_account_id',
+        'tip'       => __('You can your account ID on papercups app dashboard', $this->plugin_name)
```
Comment (reichert621) [inline admin/class-papercups-admin.php:115]: typo here as well -- let's update this to something like:
"You can get your account ID in the Papercups dashboard" (and maybe provide a link if possible?)

File: admin/class-papercups-admin.php [hunk 3]
```diff
@@ -83,51 +70,175 @@ public function enqueue_styles()
...
+      ),
+    );
+
+    add_settings_field(
```
Comment (reichert621) [inline admin/class-papercups-admin.php:119]: is there a way to mark certain fields optional? mosts of these fields will likely not need to be set, or we can just prefill them with default values

File: admin/class-papercups-admin.php (hunk 2 above)
Comment (ZmagoD) [reply admin/class-papercups-admin.php:115]: hmm I think this gets sanitised maybe 🤔

File: admin/class-papercups-admin.php (hunk 3 above)
Comment (ZmagoD) [reply admin/class-papercups-admin.php:119]: yes sure we can prefill them by adding the default `defalut` value like here
`register_setting('general', 'papercups_base_url', array('default' => 'https://app.papercups.io'));`
if you have any suggestions we can add other defaults

**Comment by ZmagoD** [comment]
@reichert621 sorry for a late response, I fixed the typos in my last commit

**Pull request merged**